pkg/bicep/schematobicep.go

```diff
@@ -36,16 +39,69 @@ func SchemaToBicep(in io.Reader) ([]byte, error) {
 }
 
 func createBicepParameter(name string, sch *schema.Schema, buf *bytes.Buffer) error {
-	bicepType, err := getBicepType(sch.Type)
+	bicepType, err := getBicepTypeFromSchema(sch.Type)
 	if err != nil {
 		return err
 	}
 
-	buf.WriteString(fmt.Sprintf("param %s %s\n", name, bicepType))
+	writeDescription(sch, buf)
+	writeAllowedParams(sch, buf)
+	writeMinValue(sch, buf, bicepType)
+	writeMaxValue(sch, buf, bicepType)
+	writeMinLength(sch, buf, bicepType)
+	writeMaxLength(sch, buf, bicepType)
+	writeSecure(sch, buf, bicepType)
+	writeBicepParam(name, sch, buf, bicepType)
 	return nil
 }
 
-func getBicepType(schemaType string) (string, error) {
+func writeBicepParam(name string, sch *schema.Schema, buf *bytes.Buffer, bicepType string) error {
+	var defVal string
+
+	if sch.Default != nil {
+		renderedVal, err := renderBicep(sch.Default, "")
+		if err != nil {
+			return err
+		}
+
+		defVal = fmt.Sprintf(" = %s", renderedVal)
+	}
+
+	buf.WriteString(fmt.Sprintf("param %s %s%s\n", name, bicepType, defVal))
+	return nil
+}
+
+func renderBicep(val interface{}, prefix string) (string, error) {
+	interfaceType, err := getBicepTypeFromInterface(val)
+	if err != nil {
+		return "", err
+	}
+
+	switch interfaceType {
```

Comment:
You could technically just turn this into the switch statement in `getBicepTypeFromInterface` (and subsequently delete that function). Its not called anywhere but here. Then its a single switch statement, instead of 2 switch statements (and a function call) for the same purpose.

So instead of `interface type -> string -> output` it would just be `interface type -> output`